bluesky/simulators.py

```diff
@@ -114,3 +121,401 @@ def check_limits(plan):
+
+    Return Parameters
+    -----------------
+    out_time : namedtuple.
+        The combined est_time/std_dev tuple.
+    """
+    
+    if method == 'sum':
+        #adds the est_time, uses quadrature for std_dev.
+        out_time = _TimeStats(est_time_1.est_time + est_time_2.est_time, 
+                        (est_time_1.std_dev**2 + est_time_2.std_dev**2)**.5)
+
+    if method == 'subtract':
+        #adds the est_time, uses quadrature for std_dev.
+        out_time = _TimeStats(est_time_1.est_time - est_time_2.est_time, 
+                        (est_time_1.std_dev**2 + est_time_2.std_dev**2)**.5)
+
+    elif method == 'max':
+        #finds the max est_time, finds the potentially largest difference from the max_time
+        #as the std_dev.
+        max_time = max(est_time1.est_time, est_time2.est_time) 
+        pos_stdev = max(est_time1.est_time + est_time1.std_dev,
+                        est_time2.est_time + est_time2.std_dev )
+        neg_stdev = min(est_time1.est_time - est_time1.std_dev,
+                        est_time2.est_time - est_time2.std_dev )
+        max_stdev = max(abs(pos_stdev-max_time), abs(neg_stdev-max_time))
+        out_time = _TimeStats(max_time, max_stdev)
+
+    return out_time
+
+
+
+def obj_est(msg ):
+    """
+    Returns the est_time/std_dev pair for the object referenced in msg.
+    This function returns the est_time/std_dev pair for the object referenced in msg.obj and 
+    for the command type reference in msg.command.
+
+    Parameters
+    ----------
+    msg : message.
+        The first message of the group that was detected in plan_ETA.
+    
+    Return Parameters
+    -----------------
+    out_time : list.
+        The combined est_time/std_dev pair.
+
+    Updated Parameters
+    ------------------
+    _plan_history : dict.
+        A dictionary containing information on values updated during the plan. It has key:arg 
+        pairs with keys relating to message components where each arg is a dictionary 
+        containing devices (detectors, axes) that have been updated, as keywords and a value 
+        indicating the update status. For 'set' the update status is the latest 'position' that
+        it is to be set and for 'trigger' the update status is the number of times since the 
+        beginning of the scan or since an 'unstage' event that the device has been triggered.
+        NOTE: Any 'set' ot 'trigger' messages processed in this function will be added to the 
+        dictionary
+    """
+
+    set_dict={}
+
+    if msg.obj is not None:
+
+        if msg.command is 'sleep':
+            return [msg.args[0], 0]
+
+        elif msg.command in (_run_start_cmds + _group_start_cmds + _timed_cmds):
+            obj =  msg.obj
+            #The if-elif section below is used to track how many 'triggers'/'sets' have 
+            #occured since the last 'unstage'/start of plan.
+            if msg.command == 'unstage':
+                _plan_history['trigger'][msg.obj.name] = 0
```

Comment:
`del _plan_history['trigger'][msg.obj.name]` might be a clearer way of indicating "not applicable" than `0`.

Reply:
This is set as zero as the 'value' is the number of triggers since the last 'unstage' not just an indication if it is staged. I did it this way in preparation for the custom devices like at CSX where some staging is done in the first trigger.